src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,529 @@
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+    if facet_dims is None:
+        facet_dims = []
+
+    # check for duplication of facet_dims in top level arg input and pc_kwargs
+    if "cols" in pc_kwargs and len(facet_dims) > 0:
+        raise ValueError(
+            f"""Facet dimensions have been defined twice.
+                Please pick only one of `facet_dims` or `pc_kwargs['cols']`.
+                Currently defined facet_dims = {facet_dims}
+                Currently defined pc_kwargs['cols'] = {pc_kwargs["cols"]}"""
+        )
+
+    if group not in ("posterior", "prior"):
+        raise TypeError("`group` argument must be either `posterior` or `prior`")
+
+    predictive_data_group = f"{group}_predictive"
+    default_observed = ""
+    if observed is None:
+        default_observed = f"(set as default for chosen group {group})"
+        observed = group == "posterior"  # by default true if posterior, false if prior
+
+    # checking if plot_kwargs["observed"] or plot_kwargs["aggregate"] is not inconsistent with
+    # top level bool arguments `observed`` and `aggregate`
+
+    # observed args logic check:
+    # observed will be True/False depending on user-input/group- true for posterior, false for prior
+    # if observed and no plot_kwargs['observed'], no prob
+    # if observed = True and plot_kwargs['observed'] = True, no prob (observed is plotted)
+    # if observed = True and plot_kwargs['observed'] = False, prob (error/warning raised)
+    # if observed = False and plot_kwargs['observed'] = True, prob (error/warning raised)
+    # if observed = False and plot_kwargs['observed'] = False, no prob (observed is not plotted)
+    if (
+        observed
+        and plot_kwargs.get("observed", True) is False
+        or not observed
+        and plot_kwargs.get("observed", False) is not False
+    ):
+        # raise warning or error
+        raise ValueError(
+            f"""
+            `observed` and `plot_kwargs["observed"]` inconsistency detected.
+            `observed` = {observed}{default_observed}
+            `plot_kwargs["observed"]` = {plot_kwargs["observed"]}
+            Please make sure `observed` and `plot_kwargs["observed"]` have the same value."""
+        )
+
+    # same check for aggregate:
+    if (
+        aggregate
+        and plot_kwargs.get("aggregate", True) is False
+        or not aggregate
+        and plot_kwargs.get("aggregate", False) is not False
+    ):
+        # raise warning or error
+        raise ValueError(
+            f"""
+            `aggregate` and `plot_kwargs["aggregate"]` inconsistency detected.
+            `aggregate` = {aggregate}
+            `plot_kwargs["aggregate"]` = {plot_kwargs["aggregate"]}
+            Please make sure `aggregate` and `plot_kwargs["aggregate"]` have the same value."""
+        )
```

Comment:
Seeing the tests and having reviewed plot_forest, I realized this is different behaviour from `plot_forest`. It might make sense to mimic the behaviour completely, which would be raising an error if `plot_kwargs["aggregate"]` (or observed) is `False`, continue on ahead otherwise.

This would mean we could have a dict in that key that gets ignored and the element is not plotted, but that is also what happens in plot_forest and I think is good enough behaviour.